### PR TITLE
Report failure in subscribing to neighbor notifications

### PR DESCRIPTION
**Pull request by aboch**
- currently it is silenced

Signed-off-by: Alessandro Boch <aboch@docker.com>

### CODE REVIEW

File: drivers/overlay/ov_network.go [hunk 1]
```diff
@@ -613,7 +613,8 @@ func (n *network) initSandbox(restore bool) error {
 	sbox.InvokeFunc(func() {
 		nlSock, err = nl.Subscribe(syscall.NETLINK_ROUTE, syscall.RTNLGRP_NEIGH)
 		if err != nil {
-			err = fmt.Errorf("failed to subscribe to neighbor group netlink messages")
+			logrus.Errorf("Failed to subscribe to neighbor group netlink messages for overlay network %s in sbox %s: %v",
+				n.id, sbox.Key(), err)
```
Comment (mavenugo) [inline drivers/overlay/ov_network.go]: Isnt it better to return the error from initSandbox in case of failure to subscribe to NEIGH miss notification ?

Comment (aboch) [inline drivers/overlay/ov_network.go]: Agree. I kept the current logic, but I do not see a reason to let the network creation go through if it won't work properly.

**Comment by aboch** [comment]
Thanks @mavenugo took care of your comment. PTAL when you get a chance.

**Comment by sanimej** [comment]
@aboch In swarm mode we don't rely on the notification. I think we should log an error but allow initSandbox to continue.

**Comment by aboch** [comment]
@santhosh Yeah, makes sense. I updated the PR. Thanks.

**Comment by sanimej** [comment]
@abcoh We are changing the check from `nlSock != nil` to `err == nil`. So please confirm `nl.Subscribe` won't return an err in some weird case but also return a valid socket. :)

LGTM

**Comment by aboch** [comment]
@sanimej Yes it does not https://github.com/vishvananda/netlink/blob/master/nl/nl_linux.go#L538

In any case we should never discard an `err != nil` even if the socket returned was not nil.

**Pull request merged**